### ISSUE TITLE
bug with the version 0.10.2 

### ISSUE THREAD

**Issue by julienlavergne** [novoda/gradle-android-test-plugin#34]
Hi,

I have a problem with the version 0.10.2 of this plugin. Below the following error under console and IntelliJ Idea:

<code>
Error:(15, 0) Gradle DSL method not found: 'toBoolean()'
Possible causes:<ul><li>The project 'ligeris-android' may be using a version of Gradle that does not contain the method.
<a href="open.wrapper.file">Open Gradle wrapper file</a></li><li>The build file may be missing a Gradle plugin.
<a href="apply.gradle.plugin">Apply Gradle plugin</a></li>
</code>

I use the correct version of the different plugins:

<code>
            ['com.android.tools.build:gradle:1.0.0'],
            ['com.neenbedankt.gradle.plugins:android-apt:1.4'],
            ['com.novoda:gradle-android-test-plugin:0.10.2']
</code>

I precise that the previous error doesn't happen when I use the versions 0.10.0 and 0.10.1 of this plugin.

**Comment by xrigau** [comment]
Hi @julienlavergne 

Thanks for raising this issue. When do you have this error? Which command are you using for running the tests?

Thanks!

**Comment by julienlavergne** [comment]
No problem. I try to participate to the OSS community.

I have this error when I run one of gradle tasks as check or assembleDebug


On February 9, 2015 at 4:53:51 PM, Xavi Rigau (notifications@github.com) wrote:

Hi @julienlavergne

Thanks for raising this issue. When do you have this error? Which command are you using for running the tests?

Thanks!

**Comment by blundell** [comment]
and what is the module `ligeris-android` what gradle version is it using?

**Comment by julienlavergne** [comment]
Ligeris-android is the name of my project.

I use gradle 2.2.1

On February 10, 2015 at 11:24:35 AM, Paul Blundell (notifications@github.com) wrote:
and what is the module ligeris-android what gradle version is it using?

**Comment by eggertj12** [comment]
I have this same issue with 0.10.2.

Starting a fresh Android Studio project. After adding plugin to module build.gradle project sync fails, by changing to 0.10.1 the sync succeeds.

I have gradle 2.2.1 on Mac, java 1.8

**Comment by julienlavergne** [comment]
@Eggert

It’s exactly my case.

On February 11, 2015 at 6:40:31 PM, Eggert Jóhannesson (notifications@github.com) wrote:

I have this same issue with 0.10.2.

Starting a fresh Android Studio project. After adding plugin to module build.gradle project sync fails, by changing to 0.10.1 the sync succeeds.

I have gradle 2.2.1 on Mac, java 1.8

**Comment by nenick** [comment]
I guess the cause is this commit https://github.com/novoda/gradle-android-test-plugin/commit/969187b9352d8341630f78979103bc1cac8f4daf

Maybe toBoolean can be replaced with just `true`

    if (project.hasProperty("testDebug")) {
        testRunTask.debug(project.testDebug.toBoolean())
    }

**Comment by nenick** [comment]
This error occur in a simple project without flavours. It can be forced if you remove following lines from the sample project.

    buildTypes {
        debug {}
        release {}
        QA {}
    }

    productFlavors {

        flavorDimensions "pricing", "releaseType"

        beta {
            flavorDimension "releaseType"
        }

        normal {
            flavorDimension "releaseType"
        }

        free {
            flavorDimension "pricing"
        }

        paid {
            flavorDimension "pricing"
        }
    }

**Comment by julienlavergne** [comment]
Thanks for your comment.

But it doesn’t seem to be a solution to modify a project with a trick.  
 

On March 2, 2015 at 10:16:08 PM, Nico Küchler (notifications@github.com) wrote:

This error occur in a simple project without flavours. It can be forced if you remove following lines from the sample project.

buildTypes {
    debug {}
    release {}
    QA {}
}

productFlavors {

    flavorDimensions "pricing", "releaseType"

    beta {
        flavorDimension "releaseType"
    }

    normal {
        flavorDimension "releaseType"
    }

    free {
        flavorDimension "pricing"
    }

    paid {
        flavorDimension "pricing"
    }
}

**Comment by nenick** [comment]
That was not a solution ;) 
The solution is to revert the commit https://github.com/novoda/gradle-android-test-plugin/commit/969187b9352d8341630f78979103bc1cac8f4daf and I hope that will be ok

**Comment by julienlavergne** [comment]
Effectively 

last release has not the bug


On March 2, 2015 at 10:29:11 PM, Nico Küchler (notifications@github.com) wrote:

That was not a solution ;)
The solution is to revert the commit 969187b and I hope that will be ok

**Comment by nenick** [comment]
Until they merge my last pull request you can use https://github.com/nenick/android-gradle-template/blob/master/Scripts/install-custom-gradle-test-plugin.sh to fix the toBoolean error. The new plugin version is 0.10.4

**Comment by julienlavergne** [comment]
Thx Nico

I dont undetstand why the merging is so long. 

Need to have several collaborators on this project to raise it.



Julien

**Comment by nenick** [comment]
It depends ;) 
My last pull request was merged within few minutes.

**Comment by neworld** [comment]
I checked using gradle 2.2.1. Everything works. All properties to project is a `String`. `String` has `toBoolean()` method: http://groovy.codehaus.org/groovy-jdk/java/lang/String.html#toBoolean()

If I understand correctly, someone override this property to something which is not the instance of `String`. Problem can be solved easily just by renaming this property. Maybe could you replace these lines with

```
        if (project.hasProperty("testDebug")) {
            println project.dump()
            System.exit(0)
            testRunTask.debug(project.testDebug.toBoolean())
        }
```

and show me?

**Comment by nenick** [comment]
Build an module without flavors and you will get the error.

**Comment by blundell** [comment]
Fixed in `0.10.4` & released https://github.com/novoda/gradle-android-test-plugin/releases/tag/0.10.4
